src/engines/http.ts

```diff
@@ -0,0 +1,155 @@
+import {
+	ConnectionUnavailable,
+	HttpConnectionError,
+	MissingNamespaceDatabase,
+} from "../errors";
+import type { RpcRequest, RpcResponse } from "../types";
+import { getIncrementalID } from "../util/getIncrementalID";
+import { retrieveRemoteVersion } from "../util/versionCheck";
+import {
+	AbstractEngine,
+	ConnectionStatus,
+	type EngineEvents,
+} from "./abstract";
+
+export class HttpEngine extends AbstractEngine {
+	connection: {
+		url?: URL;
+		namespace?: string;
+		database?: string;
+		token?: string;
+		variables: Record<string, unknown>;
+	} = { variables: {} };
```

Comment:
For performance I would initialize them with all frequently used to avoid change the shape of the object

Reply:
As in make them `undefined` by default?

Reply:
Yes